scipy/stats/_continuous_distns.py

```diff
@@ -6590,12 +6593,20 @@ def _stats(self, nu):
         return mu, mu2, g1, g2
 
     def _entropy(self, nu):
-        A = sc.gammaln(nu) - np.log(2)
-        B = -0.5 * np.log(nu)
-        C = (2 * nu - (2 * nu - 1) * sc.digamma(nu)) / 2
+        shape = np.shape(nu)
+        # because somehow this isn't taken care of by the infrastructure...
+        nu = np.atleast_1d(nu)
+        A = sc.gammaln(nu)
+        B = nu - (nu - 0.5) * sc.digamma(nu)
+        C = -0.5 * np.log(nu) - np.log(2)
         h = A + B + C
-
-        return h
+        # This is the asymptotic sum of A and B (see gh-17868)
+        norm_entropy = stats.norm._entropy()
+        # Above, this is lost to rounding error for large nu, so use the
+        # asymptotic sum when the approximation becomes accurate
+        i = nu > 1e7  # roundoff error ~ approximation error
```

Comment:
```suggestion
        i = nu > 1e5  # roundoff error ~ approximation error
```
On the left, the error is ~1e-11 and on the right, the error is ~1e-12. I don't know if it needs to be tuned much more precisely.